src/Ombi.Core/Authentication/LdapUserManager.cs

```diff
@@ -0,0 +1,168 @@
+using Novell.Directory.Ldap;
+using Ombi.Core.Settings;
+using Ombi.Settings.Settings.Models;
+using Ombi.Store.Entities;
+
+namespace Ombi.Core.Authentication
+{
+    /// <summary>
+    /// Ldap Authentication Provider
+    /// </summary>
+    public class LdapUserManager : ILdapUserManager
+    {
+        public LdapUserManager(ILogger<LdapUserManager> logger, ISettingsService<LdapSettings> ldapSettings)
+        {
+            _ldapSettingsService = ldapSettings;
+            _logger = logger;
+        }
+
+        private readonly ISettingsService<LdapSettings> _ldapSettingsService;
+        private readonly ILogger<LdapUserManager> _logger;
+
+        public async Task<LdapSettings> GetSettings()
+        {
+            return await _ldapSettingsService.GetSettingsAsync();
+        }
+
+        public async Task<OmbiUser> LdapEntryToOmbiUser(LdapEntry entry)
+        {
+            var settings = await GetSettings();
+            var userName = GetLdapAttribute(entry, settings.UsernameAttribute).StringValue;
+
+            return new OmbiUser
+            {
+                UserType = UserType.LdapUser,
+                ProviderUserId = entry.Dn,
+                UserName = userName
+            };
+        }
+
+        private LdapAttribute GetLdapAttribute(LdapEntry userEntry, string attr)
+        {
+            try
+            {
+                return userEntry.GetAttribute(attr);
+            }
+            catch (Exception)
+            {
+                return null;
+            }
+        }
+
+        private async Task<LdapConnection> BindLdapConnection(string username, string password)
+        {
+            var settings = await GetSettings();
+
+            var ldapClient = new LdapConnection { SecureSocketLayer = settings.UseSsl };
+            try
+            {
+                if (settings.SkipSslVerify)
+                {
+                    ldapClient.UserDefinedServerCertValidationDelegate += LdapClient_UserDefinedServerCertValidationDelegate;
+                }
+
+                ldapClient.Connect(settings.Hostname, settings.Port);
+                if (settings.UseStartTls)
+                {
+                    ldapClient.StartTls();
+                }
+
+                ldapClient.Bind(username, password);
+            }
+            catch (Exception e)
+            {
+                _logger.LogError(e, "Failed to Connect or Bind to server");
+                throw e;
+            }
+            finally
+            {
+                ldapClient.UserDefinedServerCertValidationDelegate -= LdapClient_UserDefinedServerCertValidationDelegate;
+            }
+
+            if (!ldapClient.Bound)
+            {
+                ldapClient.Dispose();
+                return null;
+            }
+
+            return ldapClient;
+        }
+
+        private async Task<ILdapSearchResults> SearchLdapUsers(LdapConnection ldapClient)
+        {
+            var settings = await GetSettings();
```

Comment:
We should also just pass the settings object into this method, saves another DB call